src/App.jsx

```diff
@@ -1,58 +1,51 @@
 import 'bulma/css/bulma.css';
 import './App.scss';
+import { useState, useEffect } from 'react';
+import { Button } from './components/Button';
+import { goodsFromServerModObj, buttons }
+from './helpersfuncs/getBtnGoodsObjArrs';
+import { getGoodsToRend } from './helpersfuncs/getGoodsToRend';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+export const App = () => {
+  const [searchField, setSearchField] = useState('');
+  const [goodsRender, setGoodsRender]
+  = useState(goodsFromServerModObj);
+  const [isReversed, setIsReversed] = useState(false);
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
+  const goodsToSet = getGoodsToRend(goodsFromServerModObj, searchField.btn);
 
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
+  useEffect(() => {
+    if (!isReversed) {
+      setGoodsRender(goodsToSet);
+    } else {
+      setGoodsRender(prev => [...prev].reverse());
+    }
+  }, [searchField, isReversed]);
 
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {buttons.map(btn => (
+          <Button
+            key={btn.id}
+            btn={btn.name}
+            searchField={searchField}
+            setSearchField={setSearchField}
+            goodsRender={goodsRender}
+            setGoodsRender={setGoodsRender}
+            isReversed={isReversed}
+            setIsReversed={setIsReversed}
+          >
+            { btn }
```

Comment:
if you're passing it as the children prop, then get it in the component as children